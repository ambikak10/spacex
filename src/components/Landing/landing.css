* {
  box-sizing: border-box;
}

.row {
  width: 100%;
  position: relative;
}
.selection-box {
  width: 200px;
  height: 440px;
  background-color: white;
  margin-left: 10px;
  border-radius: 5px;
  margin-top: -30px;
}
hr {
  height: 1px;
  border: 0;
  border-top: 1px solid #ccc;
  margin-left: auto;
  margin-right: auto;
  padding: 0px;
  width: 120px;
  margin-top: 2px;
}
a {
  margin-bottom: 6px;
  background-color: rgb(197, 231, 133);
  border: transparent;
  border-radius: 3px;
  height: 20px;
  width: 50px;
  font: 400 13.3333px Arial;
  appearance: button;
  text-rendering: auto;
  color: -internal-light-dark(black, white);
  letter-spacing: normal;
  word-spacing: normal;
  text-transform: none;
  text-indent: 0px;
  text-shadow: none;
  display: inline-block;
  text-align: center;
  align-items: flex-start;
  cursor: pointer;
  text-decoration: none;
  color: black;
  padding-top: 2px;
  margin-left: -3px;
}
 a:focus{
 background-color: rgb(115, 180, 9)
}
.buttons {
  margin-left: 35px;
  margin-top: -5px;
}
.mission-patch {
  max-width:100%;
  max-height:100%;
  background-color: #ececec;
  margin-top:15px;
  margin-bottom:15px;
  margin-left: auto;
  margin-right:auto;
}
.mission-patch-box {
  height: 190px;
  width: 220px;
  padding-top:5px;
  padding-left:-5px;
  padding-right:-5px;
  margin-left: auto;
  margin-right:auto;
}
/*********Large devices*********/
@media (min-width: 1440px){
  .col-xl-1,
  .col-xl-2,
  .col-xl-3,
  .col-xl-4,
  .col-xl-5,
  .col-xl-6,
  .col-xl-7,
  .col-xl-8,
  .col-xl-9,
  .col-xl-10,
  .col-xl-11,
  .col-xl-12 {
    float: left;
  }
  .col-xl-1 {
    width: 8.33%;
  }

  .col-xl-2 {
    width: 16.66%;
  }

  .col-xl-3 {
    width: 25%;
  }

  .col-xl-4 {
    width: 33%;
  }

  .col-xl-5 {
    width: 41.66%;
  }

  .col-xl-6 {
    width: 50%;
  }

  .col-xl-7 {
    width: 58.33%;
  }
  .col-xl-8 {
    width: 66.66%;
  }

  .col-xl-9 {
    width: 74.99%;
  }

  .col-xl-10 {
    width: 83.33%;
  }

  .col-xl-11 {
    width: 91.66%;
  }

  .col-xl-12 {
    width: 100%;
  }
  .program-card {
  width: 260px;
  height: 480px;
  background-color: white;
  margin-left: 10px;
  border-radius: 5px;
  margin-top: -15px;
  margin-bottom: 30px;
}
 #mission-name {
  margin-top: 60px;
  color: #0645ad;
  margin-left:18px;
 }
  #details {
  margin: 18px;
  font-size: 0.83em;
  /* display: block; */
 }
}

/***********Desktop devices*********/
@media (min-width: 1024px) and (max-width: 1440px){
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12 {
    float: left;
  }
  .col-lg-1 {
    width: 8.33%;
  }

  .col-lg-2 {
    width: 16.66%;
  }

  .col-lg-3 {
    width: 25%;
  }

  .col-lg-4 {
    width: 33%;
  }

  .col-lg-5 {
    width: 41.66%;
  }

  .col-lg-6 {
    width: 50%;
  }

  .col-lg-7 {
    width: 58.33%;
  }
  .col-lg-8 {
    width: 66.66%;
  }

  .col-lg-9 {
    width: 74.99%;
  }

  .col-lg-10 {
    width: 83.33%;
  }

  .col-lg-11 {
    width: 91.66%;
  }

  .col-lg-12 {
    width: 100%;
  }
  .program-card {
  width: 200px;
  height: 440px;
  background-color: white;
  margin-left: 90px;
  border-radius: 5px;
  margin-top: -15px;
  margin-bottom: 30px;
  margin-right: 90px;
}
 #mission-name {
  margin-top: 60px;
  color: #0645ad;
  margin-left:18px;
 }
 #details {
  margin: 18px;
  font-size: 0.83em;
}
}
/************Tablet View***********/

@media (min-width: 700px) and (max-width: 1024px) {
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12 {
    float: left;
  }

  .col-md-1 {
    width: 8.33%;
  }

  .col-md-2 {
    width: 16.66%;
  }

  .col-md-3 {
    width: 25%;
  }

  .col-md-4 {
    width: 33%;
  }

  .col-md-5 {
    width: 41.66%;
  }

  .col-md-6 {
    width: 50%;
  }

  .col-md-7 {
    width: 58.33%;
  }
  .col-md-8 {
    width: 66.66%;
  }

  .col-md-9 {
    width: 74.99%;
  }

  .col-md-10 {
    width: 83.33%;
  }

  .col-md-11 {
    width: 91.66%;
  }

  .col-md-12 {
    width: 100%;
  }
  .program-card{
  width: 240px;
  height: 400px;
  background-color: white;
  border-radius: 5px;
  margin-top: -15px;
  margin-bottom: 30px;
  }
   #mission-name {
  margin-top: 60px;
  color: #0645ad;
  margin-left:18px;
 }
}

/***********Mobile devices*********/
@media (max-width: 700px) {
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12 {
    float: left;
  }

  .col-sm-1 {
    width: 8.33%;
  }

  .col-sm-2 {
    width: 16.66%;
  }

  .col-sm-3 {
    width: 25%;
  }

  .col-sm-4 {
    width: 33%;
  }

  .col-sm-5 {
    width: 41.66%;
  }

  .col-sm-6 {
    width: 50%;
  }

  .col-sm-7 {
    width: 58.33%;
  }
  .col-sm-8 {
    width: 66.66%;
  }

  .col-sm-9 {
    width: 74.99%;
  }

  .col-sm-10 {
    width: 83.33%;
  }

  .col-sm-11 {
    width: 91.66%;
  }

  .col-sm-12 {
    width: 100%;
    text-align: center;
  }
  h2{
  text-align: center;
  }
  .program-card{
  width: 230px;
  height: 480px;
  background-color: white;
  margin-left:auto;
  margin-right:auto;
  border-radius: 5px;
  margin-top: 25px;
  margin-bottom: 10px
  }
  .buttons{
    margin-left: 8px
  }
  .selection-box {
  width: 230px;
  height: 480px;
  background-color: white;
  margin-left: auto;
  margin-right: auto;
  border-radius: 5px;
  margin-top: -10px;
}
 #mission-name {
  margin-top: 60px;
  color: #0645ad;
  margin-left:18px;
 }
}
