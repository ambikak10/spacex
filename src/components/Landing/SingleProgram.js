import React, { Component, Fragment } from 'react'
import './landing.css'

class singleProgram extends Component {
  render() {
    const {
      img,
      name,
      missionid,
      launchyear,
      launch,
      land,
    } = this.props;
  
    return (
      <Fragment>
        <section className='program-card'>
          <div className='mission-patch-box'>
            <div>
              {" "}
              <img alt="mission patch"className='mission-patch' src={img} />
            </div>
          </div>
          <h4 class='mission-name'>{name}</h4>
          <span>
            <b id='details'>Mission Ids:</b>
          </span>
          {missionid && missionid.length > 0 ? (
            missionid.map((id) => (
              <Fragment>
                {" "}
                <ul>
                  <li style={{ marginTop: "-15px", fontSize: "15px" }}>{id}</li>
                </ul>
              </Fragment>
            ))
          ) : (
            <Fragment>
              {" "}
              <span style={{ marginLeft: "-10px", fontSize: "0.83em" }}>
                
              </span>
            </Fragment>
          )}
          <Fragment>
            {" "}
            <div id='details'>
              <b>Launch Year:</b> <span>{launchyear}</span>
            </div>
          </Fragment>
          {launch === null ? (
            <div id='details'>
              <b>Successful Launch:</b> <span>NA</span>
            </div>
          ) : (
            <div id='details'>
              <b>Successful Launch:</b> <span>{launch.toString()}</span>
            </div>
          )}

          {land === null ? (
            <div id='details'>
              <b>Successful Landing:</b> <span>NA</span>
            </div>
          ) : (
            <div id='details'>
              <b>Successful Landing:</b>
              <span style={{ marginLeft: "5px" }}>{land.toString()}</span>
            </div>
          )}
        </section>
      </Fragment>
    );
  }
}

export default singleProgram

