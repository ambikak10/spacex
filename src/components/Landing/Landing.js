import React, { Component, Fragment } from "react";
import { landingPage, filter } from "../../actions/actions";
import "./landing.css";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import SingleProgram from "./SingleProgram";

class Landing extends Component {
  constructor() {
    super();
    this.state = {
      activeYear: false,
      // img: null,
      // name: null,
      // missionid: null,
      // launchyear: null,
      // launch: null,
      // land: null,
      value1: null, //launch boolean value
      value2: null, //land boolena value
      activeLaunchtrue: false,
      activeLaunchfalse: false,
      activeLandtrue: false,
      activeLandfalse: false,
      activeClass: null
    };
    this.onFilter = this.onFilter.bind(this);
  }
  componentDidMount(){
    this.props.landingPage();
  }

  onFilter (year, value1, value2, i) {
    var params = "";

    if (
      year !== null &&
      this.state.activeYear === false &&
      year !== this.state.launchyear
    ) {
      this.setState(
        {
          launchyear: year,
          activeYear: !this.state.activeYear,
          activeClass: i,
        },
        () => {
          callback();
        }
      );
    } else {
      if (
        year !== null &&
        this.state.activeYear === true &&
        year === this.state.launchyear
      ) {
        this.setState(
          {
            launchyear: null,
            activeYear: !this.state.activeYear,
            activeClass: i
          },
          () => {
            callback();
          }
        );
      } else if (
        year !== null &&
        this.state.activeYear === true &&
        year !== this.state.launchyear
      ) {
        this.setState(
          {
            launchyear: year,
            activeYear: true,
            activeClass: i,
          },
          () => {
            callback();
          }
        );
      }
    }

    //value1
    if (value1 === true){
      if(value1 !== null &&
      this.state.activeLaunchtrue === false &&
      value1 !== this.state.value1
    ) {
      this.setState(
        {
          value1: value1,
          activeLaunchtrue: !this.state.activeLaunchtrue,
          activeLaunchfalse: false
        },
        () => {
          callback();
        }
      );
    } else {
      if (
        value1 !== null &&
        this.state.activeLaunchtrue === true &&
        value1 === this.state.value1
      ) {
        this.setState(
          {
            value1: null,
            activeLaunchtrue: !this.state.activeLaunchtrue,
            activeLaunchfalse: false,
          },
          () => {
            callback();
          }
        );
      } else if (
        value1 !== null &&
        this.state.activeLaunchtrue === true &&
        value1 !== this.state.value1
      ) {
        this.setState(
          {
            value1: value1,
            activeLaunchtrue: true,
            activeLaunchfalse: false,
          },
          () => {
            callback();
          }
        );
      }
    }
  } else if(value1 === false){
    if (
      value1 !== null &&
      this.state.activeLaunchfalse === false &&
      value1 !== this.state.value1
    ) {
      this.setState(
        {
          value1: value1,
          activeLaunchfalse: !this.state.activeLaunchfalse,
          activeLaunchtrue: false,
        },
        () => {
          callback();
        }
      );
    } else {
      if (
        value1 !== null &&
        this.state.activeLaunchfalse === true &&
        value1 === this.state.value1
      ) {
        this.setState(
          {
            value1: null,
            activeLaunchfalse: !this.state.activeLaunchfalse,
            activeLaunchtrue: false,
          },
          () => {
            callback();
          }
        );
      } else if (
        value1 !== null &&
        this.state.activeLaunchfalse === true &&
        value1 !== this.state.value1
      ) {
        this.setState(
          {
            value1: value1,
            activeLaunchfalse: true,
            activeLaunchtrue: false,
          },
          () => {
            callback();
          }
        );
      }
    }
  }
  //value2
     if (value2 === true) {
       if (
         value2 !== null &&
         this.state.activeLandtrue === false &&
         value2 !== this.state.value2
       ) {
         this.setState(
           {
             value2: value2,
             activeLandtrue: !this.state.activeLandtrue,
             activeLandfalse: null,
           },
           () => {
             callback();
           }
         );
       } else {
         if (
           value2 !== null &&
           this.state.activeLandtrue === true &&
           value2 === this.state.value2
         ) {
           this.setState(
             {
               value2: null,
               activeLandtrue: !this.state.activeLandtrue,
               activeLandfalse: null,
             },
             () => {
               callback();
             }
           );
         } else if (
           value2 !== null &&
           this.state.activeLandtrue === true &&
           value2 !== this.state.value2
         ) {
           this.setState(
             {
               value2: value2,
               activeLandtrue: true,
               activeLandfalse: null,
             },
             () => {
               callback();
             }
           );
         }
       }
     } else if (value2 === false) {
       if (
         value2 !== null &&
         this.state.activeLandfalse === false &&
         value2 !== this.state.value2
       ) {
         this.setState(
           {
             value2: value2,
             activeLandfalse: !this.state.activeLandfalse,
             activeLandtrue: null,
           },
           () => {
             callback();
           }
         );
       } else {
         if (
           value2 !== null &&
           this.state.activeLandfalse === true &&
           value2 === this.state.value2
         ) {
           this.setState(
             {
               value2: null,
               activeLandfalse: !this.state.activeLandfalse,
               activeLandtrue: null,
             },
             () => {
               callback();
             }
           );
         } else if (
           value2 !== null &&
           this.state.activeLandfalse === true &&
           value2 !== this.state.value2
         ) {
           this.setState(
             {
               value2: value2,
               activeLandfalse: true,
               activeLandtrue: null,
             },
             () => {
               callback();
             }
           );
         }
       }
     }
    var callback = () => {
      console.log(this.state.launchyear);
      if (this.state.launchyear !== null) {
        console.log("inside if");
        params = params + "year=" + this.state.launchyear;
      }

      if (this.state.value1 !== null) {
        //  console.log(this.state.value1);
        params = params + "launch=" + this.state.value1;
      }
      if (this.state.value2 !== null) {
        //  console.log(this.state.value2);
        params = params + "land=" + this.state.value2;
      }
      console.log(params);
      window.history.replaceState(null, null, `/${params}`);
      this.props.filter(
        this.state.launchyear,
        this.state.value1,
        this.state.value2
      );
    };
  };
  render() {
    let content;
    let yearInfo;
    let years;
    let uniqueYears;
    const { firstlandPrograms } = this.props.firstLanding;
    const { filteredPrograms } = this.props.firstLanding;

    if (filteredPrograms === null) {
      if (firstlandPrograms !== null && firstlandPrograms.length > 0) {
        content = firstlandPrograms.map((firstlandProgram) => (
          <div className='col-lg-3 col-md-6 col-xl-3 col-sm-12'>
            <SingleProgram
              img={firstlandProgram.links.mission_patch_small}
              key={firstlandProgram.flight_number}
              name={firstlandProgram.mission_name}
              missionid={firstlandProgram.mission_id}
              launchyear={firstlandProgram.launch_year}
              launch={firstlandProgram.launch_success}
              land={firstlandProgram.rocket.first_stage.cores[0].land_success}
            />
          </div>
        ));
      }
    } else {
      if (filteredPrograms !== null && filteredPrograms.length > 0) {
        content = filteredPrograms.map((filteredProgram) => (
          <div className='col-lg-3 col-md-6 col-xl-3 col-sm-12'>
            <SingleProgram
              img={filteredProgram.links.mission_patch_small}
              key={filteredProgram.flight_number}
              name={filteredProgram.mission_name}
              missionid={filteredProgram.mission_id}
              launchyear={filteredProgram.launch_year}
              launch={filteredProgram.launch_success}
              land={filteredProgram.rocket.first_stage.cores[0].land_success}
            />
          </div>
        ));
      }
    }
    if (firstlandPrograms !== null && firstlandPrograms.length > 0) {
      years = firstlandPrograms.map((firstlandProgram) => {
        return firstlandProgram.launch_year;
      });
      uniqueYears = years.filter((year, index) => {
        return years.indexOf(year) == index;
      });

      yearInfo = uniqueYears.map((year, index) => (
        <div className='col-lg-6 col-xl-6 col-md-6 col-sm-6'>
          <button key={index}
            onClick={() => this.onFilter(year, null, null, index)}
            className={(index === this.state.activeClass) && this.state.activeYear && "selected"}
          >
            {year}
          </button>
        </div>
      ));
      //console.log(uniqueYears);
    }
    return (
      <Fragment>
        <h2 style={{ marginLeft: "10px", marginTop: "0px" }}>
          SpaceX Launch Programs
        </h2>
        <div className='row'>
          <div className='col-lg-2 col-xl-2 col-md-4 col-sm-12'>
            <section className='selection-box'>
              <p style={{ marginLeft: "10px", fontWeight: "bold" }}>Filters</p>
              <p
                style={{
                  fontSize: "12px",
                  textAlign: "center",
                  marginTop: "-10px",
                }}
              >
                Launch Year
                <hr />
              </p>
              <div className='row'>
                <div className='buttons'>{yearInfo}</div>
              </div>

              <p
                style={{
                  fontSize: "12px",
                  textAlign: "center",
                  marginTop: "-10px",
                  clear: "both",
                }}
              >
                Successful Launch
                <hr />
              </p>
              <div className='row'>
                <div className='buttons'>
                  <div
                    className='col-lg-6 col-xl-6 col-md-6 col-sm-6'
                    style={{ marginTop: "-5px" }}
                  >
                    <button
                      onClick={() => this.onFilter(null, true, null, null)}
                      className={this.state.activeLaunchtrue && "selected"}
                    >
                      True
                    </button>
                  </div>
                  <div
                    className='col-lg-6 col-xl-6 col-md-6 col-sm-6'
                    style={{ marginTop: "-5px", marginBottom: "5px" }}
                  >
                    <button
                      onClick={() => this.onFilter(null, false, null, null)}
                      className={this.state.activeLaunchfalse && "selected"}
                    >
                      False
                    </button>
                  </div>
                </div>
              </div>
              <p
                style={{
                  fontSize: "12px",
                  textAlign: "center",
                  marginTop: "20px",
                }}
              >
                Successful Landing
                <hr />
              </p>
              <div className='row'>
                <div className='buttons'>
                  <div
                    className='col-lg-6 col-xl-6 col-md-6 col-sm-6'
                    style={{ marginTop: "-5px" }}
                  >
                    <button
                      onClick={() => this.onFilter(null, null, true, null)}
                      className={this.state.activeLandtrue && "selected"}
                    >
                      True
                    </button>
                  </div>
                  <div
                    className='col-lg-6 col-xl-6 col-md-6 col-sm-6'
                    style={{ marginTop: "-5px" }}
                  >
                    <button
                      onClick={() => this.onFilter(null, null, false, null)}
                      className={this.state.activeLandfalse && "selected"}
                    >
                      False
                    </button>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div className='col-lg-10 col-xl-10 col-md-8 col-sm-12'>
            <div className='row'>{content}</div>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  firstLanding: state.firstLanding,
});

export default connect(mapStateToProps, { landingPage, filter })(Landing);
