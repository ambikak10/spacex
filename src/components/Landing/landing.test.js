import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import Landing from './Landing'
import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16"


Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();
const store = mockStore();



const props = {};

describe('Landing Component', () => {
    it('should render correctly with default props and redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Landing {...props} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should dispatch both', () => {
    const dispatch = jest.fn();
    store.dispatch = dispatch;

    const wrapper = shallow(
        <Provider store={store}>
            <Landing {...props} error={true} />
        </Provider>);
        expect(wrapper).toHaveLength(1);
    });
});