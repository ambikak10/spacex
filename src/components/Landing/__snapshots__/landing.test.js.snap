// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component should render correctly with default props and redux store 1`] = `ShallowWrapper {}`;
